hledger-lib/Hledger/Read/Common.hs

```diff
@@ -870,21 +871,25 @@ fromRawNumber raw mExp = case raw of
 
   where
     -- Outputs digit group sizes from least significant to most significant
-    groupSizes :: [DigitGrp] -> [Int]
+    groupSizes :: [DigitGrp] -> [Word8]
     groupSizes digitGrps = reverse $ case map digitGroupLength digitGrps of
       (a:b:cs) | a < b -> b:cs
       gs               -> gs
 
-    toQuantity :: DigitGrp -> DigitGrp -> (Quantity, Int)
+    toQuantity :: DigitGrp -> DigitGrp -> (Quantity, Word8)
     toQuantity preDecimalGrp postDecimalGrp = (quantity, precision)
       where
-        quantity = Decimal (fromIntegral precision)
+        quantity = Decimal precision
                            (digitGroupNumber $ preDecimalGrp <> postDecimalGrp)
         precision = digitGroupLength postDecimalGrp
 
-    applyExp :: Int -> (Decimal, Int) -> (Decimal, Int)
-    applyExp exponent (quantity, precision) =
-      (quantity * 10^^exponent, max 0 (precision - exponent))
+    applyExp :: Integer -> (Decimal, Word8) -> (Decimal, Word8)
+    applyExp exponent (quantity, precision) = (quantity * 10^^exponent, newPrecision)
+      where
+        newPrecision | precisionDiff >= 255 = maxBound
```

Comment:
what's happening here ? this case should be impossible, is that right ? If maxBound is used, that is 255, ie maxprecisionwithpoint, right ?

Reply:
Much of this got superceded when I introduced the `Precision` type.